src/lambdas/initiateDataRequest/zendeskTicketDiffersFromRequest.spec.ts

```diff
@@ -0,0 +1,383 @@
+import { getZendeskTicket } from '../../sharedServices/zendesk/getZendeskTicket'
+import { getZendeskUser } from '../../sharedServices/zendesk/getZendeskUser'
+import { DataRequestParams } from '../../types/dataRequestParams'
+import {
+  testDataRequest,
+  testDataRequestWithAllValuesSet,
+  testDataRequestWithEmptyValuesForIds
+} from '../../utils/tests/testDataRequest'
+import { ZendeskTicket } from '../../types/zendeskTicketResult'
+import { ZendeskUser } from '../../types/zendeskUser'
+import {
+  TEST_DATE_FROM,
+  TEST_DATE_TO,
+  TEST_ZENDESK_FIELD_ID_DATA_PATHS,
+  TEST_ZENDESK_FIELD_ID_DATE_FROM,
+  TEST_ZENDESK_FIELD_ID_DATE_TO,
+  TEST_ZENDESK_FIELD_ID_EVENT_IDS,
+  TEST_ZENDESK_FIELD_ID_IDENTIFIER_TYPE,
+  TEST_ZENDESK_FIELD_ID_JOURNEY_IDS,
+  TEST_ZENDESK_FIELD_ID_PII_TYPES,
+  TEST_ZENDESK_FIELD_ID_SESSION_IDS,
+  TEST_ZENDESK_FIELD_ID_USER_IDS,
+  ZENDESK_TICKET_ID_AS_NUMBER
+} from '../../utils/tests/testConstants'
+import { zendeskTicketDiffersFromRequest } from './zendeskTicketDiffersFromRequest'
+
+jest.mock('../../sharedServices/zendesk/getZendeskTicket', () => ({
+  getZendeskTicket: jest.fn()
+}))
+
+jest.mock('../../sharedServices/zendesk/getZendeskUser', () => ({
+  getZendeskUser: jest.fn()
+}))
+
+const mockGetZendeskTicket = getZendeskTicket as jest.Mock<
+  Promise<ZendeskTicket>
+>
+const mockGetZendeskUser = getZendeskUser as jest.Mock<Promise<ZendeskUser>>
+
+describe('match zendesk ticket details', () => {
```

Comment:
We are missing the scenario where the ticket does not exist. Could you please add that?